### PR TITLE
Bug 1822119 - Store new UUID in shared prefs and send in metrics ping

### PR DESCRIPTION
**Pull request by hackebrot**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1822119

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 1]
```diff
@@ -719,6 +719,12 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
 
             searchWidgetInstalled.set(settings.searchWidgetInstalled)
 
+            if (settings.sharedPrefsUUID.isEmpty()) {
+                settings.sharedPrefsUUID = sharedPrefsUuid.generateAndSet().toString()
+            } else {
```
Comment (rocketsroger) [inline fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt:724]: Do we need else? Should `sharedPrefUuid.set()` always called

Comment (hackebrot) [reply fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt:724]: We want to see if the value changes over time. I set the `lifetime` for the new `sharedPrefsUuid` Glean metric as `ping`, so reading the value from shared prefs and setting that for the metric makes sense to me.

Does that sound reasonable to you?

Comment (rocketsroger) [reply fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt:724]: Yes, as discussed `generateAndSet()` does also set the metrics value for the ping.  Your implementation here is correct.

**Review by boek — COMMENTED** [review]
Just one question. Nothing else really jumps out to me!

**Comment by hackebrot** [comment]
Hi @rocketsroger & @boek! 👋🏻 

I have requested data collection review on [bug 1822119](https://bugzilla.mozilla.org/show_bug.cgi?id=1822119#c3) and marked this pull request as Ready for Review. I've updated the commit to now reset the `sharedPrefsUUID` value in Shared Preferences on telemetry data collection opt-out and added an inline comment for context.

Please take another look and let me know if there's something left to do before we can merge and release this change.

Thanks! 🦊

File: fenix/app/metrics.yaml [hunk 2]
```diff
@@ -1916,6 +1916,26 @@ metrics:
     metadata:
       tags:
         - Notifications
+  shared_prefs_uuid:
```
[the remark this answers is not in the document]
Comment (travis79) [reply fenix/app/metrics.yaml:2294]: @boek Recent data-review escalations have concluded that random identifiers that do not contain or are based on user information are to be classified as technical data. The Data Collection wiki should be soon updated to reflect this to make it more clear.

**Comment by rocketsroger** [comment]
hi @hackebrot please take a look at https://github.com/mozilla/data-review/blob/main/request.md for data review.  [Here](https://github.com/mozilla-mobile/firefox-android/pull/1674) is a good example as well.  Thanks

File: fenix/app/metrics.yaml [hunk 3]
```diff
@@ -2291,6 +2291,26 @@ metrics:
...
+    lifetime: ping
+    description: |
+      A UUID stored in Shared Preferences used to analyze technical differences
+      between storage mechanisms in Android, specifically the Glean DB and
+      Shared Preferences.
+    send_in_pings:
+      - metrics
+    notification_emails:
+      - android-probes@mozilla.com
+      - raphael@mozilla.com
```
Comment (rocketsroger) [inline fenix/app/metrics.yaml:2305]: 🔍 nit: for non-permanent collection `android-probes@mozilla.com` is enough.

File: fenix/app/metrics.yaml [hunk 4]
```diff
@@ -2291,6 +2291,26 @@ metrics:
...
+      - fbertsch@mozilla.com
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1822119
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1822119
+    data_sensitivity:
+      - technical
+    expires: 116
```
Comment (rocketsroger) [inline fenix/app/metrics.yaml:2313]: 🔍 nit: This will expire fairly soon.  Are we sure that version 116 is good enough?

**Review by rocketsroger — APPROVED** [review]
Looks good.  A couple questions.  Thanks

File: fenix/app/metrics.yaml (hunk 4 above)
Comment (hackebrot) [reply fenix/app/metrics.yaml:2313]: My understanding is that we will have data from Firefox 113, 114, 115.

What's your recommendation, @rocketsroger?

What do you think @Dexterp37 and @fbertsch?

Comment (Dexterp37) [reply fenix/app/metrics.yaml:2313]: No strong opinion, the decision should come from the person looking at the data , so likely you or @fbertsch?

Comment (fbertsch) [reply fenix/app/metrics.yaml:2313]: Should be easy to reenable if we want more data. Let's keep it.

**Comment by hackebrot** [comment]
Thanks for the review, @rocketsroger! Much appreciated.

What are the next steps for getting this released?

**Comment by rocketsroger** [comment]
@Mergifyio backport releases_v113

**Pull request merged**